### ISSUE TITLE
Make builds work with manage_closure_dependencies

### ISSUE THREAD

**Issue by tschaub** [openlayers/openlayers#4945]
Creating builds with `manage_closure_dependencies: true` can result in smaller build sizes.  While this doesn't seem like it should be the case, there is unused code that is unconditionally included in the builds (see #3447 for example).  Using `manage_closure_dependencies` improves the situation by limiting the files that are included for compilation.  However, because our `olx.js` file relies on types defined throughout the library, the `olx.js` file cannot safely be passed to the compiler when `manage_closure_dependencies` is set to `true`.

This can be demonstrated by creating a build config that only exports a single source (e.g. `ol.source.Raster`).  This generates compiler errors like the following:

```
externs/olx.js:2409: ERROR - Bad type annotation. Unknown type ol.interaction.DragBoxEndConditionType
 *     boxEndCondition: (ol.interaction.DragBoxEndConditionType|undefined)}}
                         ^

externs/olx.js:2446: ERROR - Bad type annotation. Unknown type ol.interaction.DragBoxEndConditionType
 * @type {ol.interaction.DragBoxEndConditionType|undefined}
          ^

2 error(s), 0 warning(s)
```

One alternative could be to distribute the externs files around the `src` directory (e.g. `src/ol/interaction/dragboxinteraction.externs.js`) and have a build tool concatenate what is needed for a custom build - but this is piling more custom logic and complexity into the build tool.

It would be nice if there were an annotation we could used to describe our externs, rather than needing to include them in separate files.  A custom annotation and build step to generate externs is another option (similar to the one above).

Anybody else have any ideas?

**Comment by ahocevar** [comment]
I don't have any alternative that would not include custom build steps. Maybe @fredj knows something?

**Comment by probins** [comment]
I thought that moving the typedefs into 1 file would fix this problem, but it doesn't. It fixes the issues for typedefs, such as `ol.interaction.DragBoxEndConditionType` mentioned, but the compile still fails with `Unknown type ol.layer.VectorTileRenderType`, which is an enum not a typedef. These could be moved into one file as well, but that still wouldn't fix the problem, as there are still constructor types referred to in olx, and these have to remain in their own module/file.

I encountered the opposite problem when I first started creating custom builds. I thought it would be really clever to only include those olx externs that I actually needed, but this too fails, as the code refers to other olx types that I don't use. I suppose the basic issue is that the compiler has to know about _all_ the type definitions, whether they're needed for a particular build or not.
